File: zema_store_backend/src/models/subscription.ts
import mongoose, { Schema, Document } from "mongoose";

export interface ISubscriptionDocument extends Document {
  subscriptionType: String;
  subscriptionId: String;
  summary: String;
}

const subscriptionSchema: Schema<ISubscriptionDocument> = new Schema({
  subscriptionType: {
    type: String,
    required: true,
  },
  subscriptionId: {
    type: String,
    required: true,
  },
  summary: {
    type: String,
    required: true,
  },
});

const Subscription = mongoose.model<ISubscriptionDocument>(
  "Subscription",
  subscriptionSchema
);

export default Subscription;

File: zema_store_mobile/lib/models/subscription.dart
import 'dart:ffi';

class Subscription{

  Subscription({
    required this.subscriptionType,
    required this.subscriptionId,
    required this.summary,
    required this.price
  });
  final String subscriptionType;
  final String subscriptionId;
  final String summary;
  final Double price;

  factory Subscription.fromJson(Map<String,dynamic> json){
    return Subscription(
      subscriptionType: json['subscriptionType'], 
      subscriptionId: json['subscriptionId'], 
      summary: json['summary'], 
      price: json['price']
      );
  }
}